### PR TITLE
Comments

### PR DESCRIPTION
**Pull request by AniMachaidze**
Experimenting with JSON and JavaScript by following the tutorial

### CODE REVIEW

File: portfolio/src/main/webapp/index.html [hunk 1]
```diff
@@ -58,13 +58,27 @@ <h1 class="title">Ani Machaidze</h1>
...
+                <form action="/data" method="POST">
+
+                    <p>Enter your name:</p>
+                    <textarea name="author"></textarea>
+                    <br/>
+                    <p>Enter your comment:</p>
+                    <textarea name="text"></textarea>
+                    <br/>
+
+                <input type="submit" />
```
Comment (astrelni) [inline portfolio/src/main/webapp/index.html:73]: I think you'll want to add a `value` attribute as well for the input button to not be empty.

File: portfolio/src/main/webapp/script.js [hunk 2]
```diff
@@ -12,9 +12,33 @@
...
-  fetch('/data').then(response => response.text()).then((quote) => {
-    document.getElementById('quote-container').innerText = quote;
+    console.log(comment.comments);
+    const comentListElement = document.getElementById('comments-container');
+	var i;
+    comentListElement.innerHTML = '';
+    
+    // gets each object from json list and appends as a <li> item
+    for (i in comment.comments) {
+        console.log(com);
```
Comment (astrelni) [inline portfolio/src/main/webapp/script.js:30]: This also seems like an old debugging line that should be removed?

File: portfolio/src/main/webapp/script.js [hunk 3]
```diff
@@ -12,9 +12,33 @@
...
+ */
+function getComments() {	
+  	fetch('/data').then(response => response.json()).then((comment) => {
+    // stats is an object, not a string, so we have to
+    // reference its fields to create HTML content
 
-function getRandomMessage() {
-  fetch('/data').then(response => response.text()).then((quote) => {
-    document.getElementById('quote-container').innerText = quote;
+    console.log(comment.comments);
```
Comment (astrelni) [inline portfolio/src/main/webapp/script.js:23]: Is this leftover debugging that should be removed?

File: portfolio/src/main/webapp/script.js [hunk 4]
```diff
@@ -12,9 +12,33 @@
...
+    const comentListElement = document.getElementById('comments-container');
```
Comment (astrelni) [inline portfolio/src/main/webapp/script.js:24]: Small typo, missing a second m in `comment`.

File: portfolio/src/main/webapp/script.js [hunk 5]
```diff
@@ -12,9 +12,33 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+/**
+ * Fetches stats from the servers and adds them to the DOM.
+ */
+function getComments() {	
+  	fetch('/data').then(response => response.json()).then((comment) => {
+    // stats is an object, not a string, so we have to
```
Comment (astrelni) [inline portfolio/src/main/webapp/script.js:20]: `stats`?

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 6]
```diff
@@ -29,34 +29,48 @@
...
+    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        // Get the input from the form.
+        String author = getParameter(request, "author", "unknown");
+        String text = getParameter(request, "text", "");
 
-        	// Send the JSON as the response
-            response.setContentType("application/json;");
-        	response.getWriter().println(jsonComment);
-        }
+        // creates new comment object and puts into list
```
Comment (astrelni) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:65]: Generally it is ok to leave a comment like this out, the line below explains itself. 

Usually we try to add comments about why something is done as opposed to what is happening. If we have to comment and what is happening that is also a sign that the code is complex enough that it belongs in its own function with a descriptive name.

File: portfolio/src/main/webapp/index.html (hunk 1 above)
Comment (astrelni) [reply portfolio/src/main/webapp/index.html:73]: Oh nevermind, I didn't see it gets the default "Submit" if you don't add anything.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 6 above)
Comment (astrelni) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Just to make sure I'm not sounding confusing, I am talking about just the code snippets here. For a comment on top of a function it's the opposite. There the comments should say what the function does, as opposed to how or why it should be used.

File: portfolio/src/main/webapp/script.js [hunk 7]
```diff
@@ -12,9 +12,28 @@
...
+	fetch('/data').then(response => response.json()).then((comment) => {
 
-function getRandomMessage() {
-  fetch('/data').then(response => response.text()).then((quote) => {
-    document.getElementById('quote-container').innerText = quote;
-  });
-}
\ No newline at end of file
+		const commentListElement = document.getElementById('comments-container');
+		var i;
```
Comment (oydang) [inline portfolio/src/main/webapp/script.js:22]: Nit: remove this line.

File: portfolio/src/main/webapp/script.js [hunk 8]
```diff
@@ -12,9 +12,28 @@
...
+		commentListElement.innerHTML = '';
+
+		for (i in comment.comments) {
```
Comment (oydang) [inline portfolio/src/main/webapp/script.js:25]: Nit: `for (var i in comment.comments) {`

**Pull request merged**